homeassistant/components/vizio/media_player.py

```diff
@@ -83,19 +98,44 @@
             "is valid and available, device type is correct%s",
             fail_auth_msg,
         )
-        return
+        raise PlatformNotReady
+
+    entity = VizioDevice(device, name, volume_step, device_type, unique_id)
+    entry.add_update_listener(async_entry_updated)
```

Comment:
Add the options update listener inside `async_added_to_hass` entity coroutine method. Store the return value as an instance attribute so that we can unsubscribe when removing the entity in `async_will_remove_from_hass`.

Reply:
Can I store the `ConfigEntry` in `VizioDevice`? If not, how do I access the `add_update_listener` function?

Reply:
Pass the config entry to the entity and store it as an instance attribute.

Reply:
Nevermind, I was able to get to it via `self.hass.config_entries.async_get_entry(self.registry_entry.config_entry_id)`